src/main/java/org/testng/internal/annotations/AnnotationHelper.java

```diff
@@ -217,7 +219,7 @@ private static void finishInitialize(ConfigurationAnnotation result, IConfigurat
 //    for (Class<?> cls : classes) {
         while (null != cls) {
           boolean hasClassAnnotation = isAnnotationPresent(annotationFinder, cls, annotationClass);
-          Method[] methods = cls.getDeclaredMethods();
+          Method[] methods = getDeclaredMethods(cls);
```

Comment:
As far as I can tell it only returns default methods if you call it on the interface, otherwise you have to do some mining. That is how it works for me on 1.8.0_72, if that's the case nothing will have to be done (unless they change it). Spring seems to prefer to call declared + default: local, is getLocalMethods agreeable?